pkg/types/interfaces.go

```diff
@@ -42,6 +42,9 @@ type Strip interface {
 type Secured interface {
 	Encrypt(context.Context, func(context.Context, []byte) ([]byte, error)) error
 	Decrypt(context.Context, func(context.Context, []byte) ([]byte, error)) error
+	IntegralData() []byte
```

Comment:
The following 3 methods are only used in isolation from the above 2, so I will move the new ones in different interface. This will allow us to check integrity for example on visibilities also. In case the labels are changed outside application knowing.